picasso/src/main/java/com/squareup/picasso3/BitmapHunter.java

```diff
@@ -388,7 +371,9 @@ static BitmapHunter forRequest(Picasso picasso, Dispatcher dispatcher,
     return new BitmapHunter(picasso, dispatcher, cache, stats, action, ERRORING_HANDLER);
   }
 
-  static Result applyCustomTransformations(List<Transformation> transformations, Result result) {
+  static Result applyTransformations(Picasso picasso, Request data,
+      List<Transformation> transformations,
+      Result result) {
     if (!result.hasBitmap()) {
```

Comment:
How many people call this method? Can we move this outside and avoid allocations? Can be follow-up.

Reply:
Allocations == new ArrayList

Reply:
removing call to result.hasBitmap from here, as it's already called outside in its sole call site: BitmapHunter#hunt.